### PR TITLE
Throw exception if IoT device module invokes an IoT edge module API

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

**Review by tmahmood-microsoft — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

File: iothub/device/src/IotHubModuleClient.cs [hunk 1]
```diff
@@ -248,6 +249,13 @@ public Task<DirectMethodResponse> InvokeMethodAsync(string deviceId, EdgeModuleD
         {
             Argument.AssertNotNullOrWhiteSpace(deviceId, nameof(deviceId));
             Argument.AssertNotNull(methodRequest, nameof(methodRequest));
+
+            if (!IotHubConnectionCredentials.IsEdgeModule)
+            {
+                throw new InvalidOperationException("This API call is relevant only for IoT Edge modules. Please make sure your client is initialized correctly." +
+                    "For IoT device module direct method invocations, see SetDirectMethodCallbackAsync(...).");
```
Comment (drwill-ms) [inline iothub/device/src/IotHubModuleClient.cs:256]: Needs a space after the period.

File: iothub/device/src/IotHubModuleClient.cs [hunk 2]
```diff
@@ -283,6 +292,13 @@ public Task<DirectMethodResponse> InvokeMethodAsync(string deviceId, string modu
             Argument.AssertNotNullOrWhiteSpace(deviceId, nameof(deviceId));
             Argument.AssertNotNullOrWhiteSpace(moduleId, nameof(moduleId));
             Argument.AssertNotNull(methodRequest, nameof(methodRequest));
+
+            if (!IotHubConnectionCredentials.IsEdgeModule)
+            {
+                throw new InvalidOperationException("This API call is relevant only for IoT Edge modules. Please make sure your client is initialized correctly." +
```
Comment (drwill-ms) [inline iothub/device/src/IotHubModuleClient.cs:298]: Can we find a way not to repeat the error message?

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**